### ISSUE TITLE
EduVPN client doesn't seem to be capturing DNS requests from systemd-resolved

### ISSUE THREAD

**Issue by GentleInfant** [eduvpn/python-eduvpn-client#566]
# Installation

Via the Ubuntu apt repository I think.

# Version

4.2.0-1+ubuntu+23.04+1

# OS/Distribution

Ubuntu 22.04.03 LTS

# Logs

Sorry don't have those handy at present! I will need to add those to this ticket later.

## Do you have a problem during or after connecting?

The issue was after connecting.

So the situation that caused it to happen is that I have a Matrix server running on my home network. In order so that I can connect to it from machines on my home network as well as out in the wild I have a rule on my router that resolves requests to `mydomain.com` to the local server hosting the service.

I noticed that when I turned EduVPN on the connection to the matrix server would break. EduVPN is set "Allow LAN Traffic" to off  and "All Traffic" to true but from some investigation it looks as though `systemd-resolved` is using the local routers entry for the server, but when it tries to connect of course that IP address doesn't exist on the uni's network so it fails.

I did capture some output. This is the output of `resolvectl` when the VPN is active
![image](https://github.com/eduvpn/python-eduvpn-client/assets/123903177/f6c5294a-1efb-4000-8972-381c38033083)

With EduVPN enabled this is comparing the `resolvectl` output for a machine on my home network vs one on the uni network 
![image](https://github.com/eduvpn/python-eduvpn-client/assets/123903177/933798c4-348a-4dcd-85a3-552e6e709096)
My feeling is that querying `mercury` should fail...?

Clearing `systemd-resolved`'s cache didn't seem to affect anything. 

After reading https://systemd.io/RESOLVED-VPNS/ I wondered if there is a setting awry on the EduVPN server side and it's acting as a corporate VPN rather than a privacy VPN despite all the settings I've used? It's getting a little beyond my paygrade at this point though!

So in summary

# What I expect

When I enable EduVPN the resolution of `mydomain.com` should be to the internet visible external IP address

# What happens

`mydomain.com` resolves to a local IP address when EduVPN is on, "All Traffic" is active, and "Allow LAN Traffic" is inactive.

**Comment by jwijenbergh** [comment]
Thanks for the issue. I would have to look into this in more details, but a preliminary question:

- Is this OpenVPN or WireGuard, it should show in the connection info when you are connected

**Comment by jwijenbergh** [comment]
Should be fixed in the latest release. If not please re-open

**Comment by GentleInfant** [comment]
Hi jwijenbergh, sorry I didn't get back to you earlier! Had some days off. Thanks for resolving it so quickly though :)

FWIW the value for `connection.type` is `wireguard`.

```
$ nmcli con show eduVPN
connection.id:                          eduVPN
connection.uuid:                        b8566d08-06d7-4eda-bf14-f1e33f382f98
connection.stable-id:                   --
connection.type:                        wireguard
connection.interface-name:              eduVPN
connection.autoconnect:                 no
connection.autoconnect-priority:        0
connection.autoconnect-retries:         -1 (default)
connection.multi-connect:               0 (default)
connection.auth-retries:                -1
connection.timestamp:                   1707255337
connection.read-only:                   no
connection.permissions:                 user:user123
connection.zone:                        --
connection.master:                      --
connection.slave-type:                  --
connection.autoconnect-slaves:          -1 (default)
connection.secondaries:                 --
connection.gateway-ping-timeout:        0
connection.metered:                     unknown
connection.lldp:                        default
connection.mdns:                        -1 (default)
connection.llmnr:                       -1 (default)
connection.dns-over-tls:                -1 (default)
connection.wait-device-timeout:         -1
ipv4.method:                            manual
ipv4.dns:                               10.64.0.112,10.64.0.232
ipv4.dns-search:                        otago.ac.nz
ipv4.dns-options:                       --
ipv4.dns-priority:                      0
ipv4.addresses:                         10.96.143.37/24
ipv4.gateway:                           --
ipv4.routes:                            --
ipv4.route-metric:                      -1
ipv4.route-table:                       51860
ipv4.routing-rules:                     priority 1 not from 0.0.0.0/0 fwmark 0xca94 table 51860
ipv4.ignore-auto-routes:                no
ipv4.ignore-auto-dns:                   no
ipv4.dhcp-client-id:                    --
ipv4.dhcp-iaid:                         --
ipv4.dhcp-timeout:                      0 (default)
ipv4.dhcp-send-hostname:                yes
ipv4.dhcp-hostname:                     --
ipv4.dhcp-fqdn:                         --
ipv4.dhcp-hostname-flags:               0x0 (none)
ipv4.never-default:                     no
ipv4.may-fail:                          yes
ipv4.required-timeout:                  -1 (default)
ipv4.dad-timeout:                       -1 (default)
ipv4.dhcp-vendor-class-identifier:      --
```

Figured it might be worth adding to the ticket after the fact in case anyone else is looking for it.

**Comment by jwijenbergh** [comment]
Thanks for the additional info. So you can confirm this was fixed in the latest release?

**Comment by GentleInfant** [comment]
I haven't been able to verify that yet sorry. I'm physically back in the office. Watch this space though!
